extensions/bear/src/index.tsx

```diff
@@ -13,18 +14,26 @@ export default function SearchNotes(props: LaunchProps<{ arguments: SearchNotesA
   const [searchQuery, setSearchQuery] = useState(initialSearchQuery ?? "");
   const [db, error] = useBearDb();
   const [notes, setNotes] = useState<Note[]>();
+  const [selectedTag, setSelectedTag] = useState<string>("");
 
   useEffect(() => {
     if (db != null) {
-      setNotes(db.getNotes(searchQuery));
+      setNotes(db.getNotes(searchQuery, selectedTag));
     }
-  }, [db, searchQuery]);
+  }, [db, searchQuery, selectedTag]);
 
   if (error) {
     showToast(Toast.Style.Failure, "Something went wrong", error.message);
   }
 
   const showDetail = (notes ?? []).length > 0 && getPreferenceValues().showPreviewInListView;
+  const handleTagChange = (tag: string) => {
+    let formattedTag = "";
+    if (tag !== formattedTag) {
+      formattedTag = tag.includes(" ") ? `#${tag}#` : `#${tag}`;
+    }
+    setSelectedTag(formattedTag);
+  };
```

Comment:
Given we've switched to joining against the tags table rather than searching within the note's content, we can go back to using the tag name without the `#` signs.

```suggestion
  const handleTagChange = (tag: string) => setSelectedTag(tag);
```